src/utils/api.js

```diff
@@ -27,23 +27,41 @@ class Api {
 		}
 	}
 
-	_refreshToken() {
+	refreshToken() {
 		if (this.isRefreshing) {
-			let bindedResolve;
-			const _processRefreshTokenEvent = (resolve, e) => {
-				window.removeEventListener(this.tokenRefreshedEventType, bindedResolve, false);
-				resolve(e.detail);
-			};
-			return new Promise((resolve) => {
-				bindedResolve = _processRefreshTokenEvent.bind(null, resolve);
-				window.addEventListener(this.tokenRefreshedEventType, bindedResolve, false);
+			const deferedPromise = {};
+			deferedPromise.p = new Promise((resolve, reject) => {
+				deferedPromise.resolve = resolve;
+				deferedPromise.reject = reject;
 			});
+			const _processRefreshTokenEvent = (e) => {
+				window.removeEventListener(this.tokenRefreshedEventType, deferedPromise, false);
+				if (e.detail && e.detail.err) {
+					deferedPromise.reject(e.detail.err);
+				} else {
+					deferedPromise.resolve(e.detail);
+				}
+			};
+			window.addEventListener(this.tokenRefreshedEventType, _processRefreshTokenEvent, false);
+			return deferedPromise.p;
 		}
 
 		this.isRefreshing = true;
 		return fetch(`${this.config.AUTH_SERVER}/api/v2/refresh_token`, {
 			method: 'POST',
 			credentials: 'include',
+		}).then((res) => {
+			this.isRefreshing = false;
+			window.dispatchEvent(new CustomEvent(this.tokenRefreshedEventType, {
+				detail: res,
+			}));
+			return res;
+		}).catch((err) => {
+			this.isRefreshing = false;
+			window.dispatchEvent(new CustomEvent(this.tokenRefreshedEventType, {
+				detail: { err },
+			}));
+			throw err;
```

Comment:
@fcjr could you move the two `this.isRefreshing = false` to just one in a `.then()` after the `.catch()`?